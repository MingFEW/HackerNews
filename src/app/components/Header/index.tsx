import React from 'react'
import 'twin.macro'

export const Header: React.FC = () => {
  return (
    <div tw="fixed w-screen backdrop-blur z-[2] border-b border-solid border-white/20">
      <div className="header" tw="px-6 py-4 mx-auto md:max-w-[640px] md:px-0 lg:max-w-[768px]">
        <h1 tw="text-white text-xl font-bold">Hacker Top Stories</h1>
        <p tw="text-textGrey text-xs">Made by MingFEW</p>
      </div>
    </div>
  )
}
