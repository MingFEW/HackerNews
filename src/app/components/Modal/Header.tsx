import React from 'react'
import { BsArrowLeft } from 'react-icons/bs'
import tw from 'twin.macro'

interface Props {
  title: string
  onClose: () => void
}

const Header: React.FC<Props> = props => {
  const { title, onClose } = props
  return (
    <div css={[tw`bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500`, tw`flex items-center gap-4 p-4`]}>
      <button type="button" tw="cursor-pointer" onClick={onClose}>
        <BsArrowLeft tw="text-white text-2xl" />
      </button>
      <div tw="text-lg font-bold text-white">{title}</div>
    </div>
  )
}

export default Header
