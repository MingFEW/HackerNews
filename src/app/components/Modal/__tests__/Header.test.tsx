import { createRenderer } from 'react-test-renderer/shallow'

import Header from '../Header'

const shallowRenderer = createRenderer()

describe('<Header />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(
      <Header
        title="story"
        onClose={() => {
          // eslint-disable-next-line no-console
          console.log('click')
        }}
      />
    )
    const renderedOutput = shallowRenderer.getRenderOutput()
    expect(renderedOutput).toMatchSnapshot()
  })
})
