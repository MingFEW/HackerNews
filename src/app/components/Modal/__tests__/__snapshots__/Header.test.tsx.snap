// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render and match the snapshot 1`] = `
<div
  tw="flex items-center gap-4 p-4 bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500"
>
  <button
    onClick={[Function]}
    tw="cursor-pointer"
    type="button"
  >
    <BsArrowLeft
      tw="text-white text-2xl"
    />
  </button>
  <div
    tw="text-lg font-bold text-white"
  >
    story
  </div>
</div>
`;
